controllers/vspherevm_controller.go

```diff
@@ -275,6 +269,15 @@ func (r vmReconciler) Reconcile(ctx goctx.Context, req ctrl.Request) (_ ctrl.Res
 		}
 	}
 
+	clusterModuleInfo, err := r.fetchClusterModuleInfo(vsphereCluster, machine)
+	// If cluster module information cannot be fetched for a VM being deleted,
+	// we should not block VM deletion since the cluster module is updated
+	// once the VM gets removed.
+	if err != nil && vsphereVM.ObjectMeta.DeletionTimestamp.IsZero() {
+		return reconcile.Result{}, err
+	}
+	vmContext.ClusterModuleInfo = clusterModuleInfo
```

Comment:
If this is nil on deletion because of the `MachineSet` being gone, I assume we're not removing the VM from the clusterModule? If so does it have any impact when trying to delete the the whole cluster and hence the cluster modules?